### ISSUE TITLE
Components should block for at most 20-30ms.

### ISSUE THREAD

**Issue by marcinbc** [mb-software/esphome-huawei-r4850#1]
Hello, thank you for your work. I have a problem with the component. I use wemos d1 + mcp251, the reading is after the start but after a few seconds I have ad and no sensor readings. update_interval extension: 20s does not bring good results
[08:50:30][D][huawei_r4850:049]: Sending request message
[08:50:30][D][sensor:093]: 'r4850 Input power': Sending state nan W with 1 decimals of accuracy
[08:50:30][D][sensor:093]: 'r4850 Input voltage': Sending state nan V with 1 decimals of accuracy
[08:50:30][D][sensor:093]: 'r4850 Input current': Sending state nan A with 2 decimals of accuracy
[08:50:30][D][sensor:093]: 'r4850 Input temperature': Sending state nan °C with 1 decimals of accuracy
[08:50:30][D][sensor:093]: 'r4850 Input frequency': Sending state nan Hz with 3 decimals of accuracy
[08:50:30][D][sensor:093]: 'r4850 Output power': Sending state nan W with 1 decimals of accuracy
[08:50:30][D][sensor:093]: 'r4850 Output current': Sending state nan A with 2 decimals of accuracy
[08:50:30][D][sensor:093]: 'r4850 Output voltage': Sending state nan V with 2 decimals of accuracy
[08:50:30][D][sensor:093]: 'r4850 Output temperature': Sending state nan °C with 1 decimals of accuracy
[08:50:30][D][sensor:093]: 'r4850 Efficiency': Sending state nan % with 0 decimals of accuracy
[08:50:30][D][number:012]: 'r4850 Max output current': Sending state nan
[08:50:30][W][component:204]: Component huawei_r4850 took a long time for an operation (0.09 s).
[08:50:30][W][component:205]: Components should block for at most 20-30ms.
[08:50:30][D][sensor:093]: 'r4850 Input power': Sending state 0.00000 W with 1 decimals of accuracy
[08:50:50][D][huawei_r4850:049]: Sending request message
[08:50:50][D][sensor:093]: 'r4850 Input power': Sending state nan W with 1 decimals of accuracy
[08:50:50][D][sensor:093]: 'r4850 Input voltage': Sending state nan V with 1 decimals of accuracy
[08:50:50][D][sensor:093]: 'r4850 Input current': Sending state nan A with 2 decimals of accuracy
[08:50:50][D][sensor:093]: 'r4850 Input temperature': Sending state nan °C with 1 decimals of accuracy
[08:50:50][D][sensor:093]: 'r4850 Input frequency': Sending state nan Hz with 3 decimals of accuracy
[08:50:50][D][sensor:093]: 'r4850 Output power': Sending state nan W with 1 decimals of accuracy
[08:50:50][D][sensor:093]: 'r4850 Output current': Sending state nan A with 2 decimals of accuracy
[08:50:50][D][sensor:093]: 'r4850 Output voltage': Sending state nan V with 2 decimals of accuracy
[08:50:50][D][sensor:093]: 'r4850 Output temperature': Sending state nan °C with 1 decimals of accuracy
[08:50:50][D][sensor:093]: 'r4850 Efficiency': Sending state nan % with 0 decimals of accuracy
[08:50:50][D][number:012]: 'r4850 Max output current': Sending state nan
[08:50:50][W][component:204]: Component huawei_r4850 took a long time for an operation (0.09 s).
[08:50:50][W][component:205]: Components should block for at most 20-30ms.
[08:50:50][D][sensor:093]: 'r4850 Input power': Sending state 0.00000 W with 1 decimals of accuracy
[08:51:10][D][huawei_r4850:049]: Sending request message
[08:51:10][D][sensor:093]: 'r4850 Input power': Sending state nan W with 1 decimals of accuracy
[08:51:10][D][sensor:093]: 'r4850 Input voltage': Sending state nan V with 1 decimals of accuracy
[08:51:10][D][sensor:093]: 'r4850 Input current': Sending state nan A with 2 decimals of accuracy
[08:51:10][D][sensor:093]: 'r4850 Input temperature': Sending state nan °C with 1 decimals of accuracy
[08:51:10][D][sensor:093]: 'r4850 Input frequency': Sending state nan Hz with 3 decimals of accuracy
[08:51:10][D][sensor:093]: 'r4850 Output power': Sending state nan W with 1 decimals of accuracy
[08:51:10][D][sensor:093]: 'r4850 Output current': Sending state nan A with 2 decimals of accuracy
[08:51:10][D][sensor:093]: 'r4850 Output voltage': Sending state nan V with 2 decimals of accuracy
[08:51:10][D][sensor:093]: 'r4850 Output temperature': Sending state nan °C with 1 decimals of accuracy
[08:51:10][D][sensor:093]: 'r4850 Efficiency': Sending state nan % with 0 decimals of accuracy
[08:51:10][D][number:012]: 'r4850 Max output current': Sending state nan
[08:51:10][W][component:204]: Component huawei_r4850 took a long time for an operation (0.09 s).
[08:51:10][W][component:205]: Components should block for at most 20-30ms.
[08:51:10][D][sensor:093]: 'r4850 Input power': Sending state 0.00000 W with 1 decimals of accuracy
[08:51:30][D][huawei_r4850:049]: Sending request message
[08:51:30][D][sensor:093]: 'r4850 Input power': Sending state nan W with 1 decimals of accuracy
[08:51:30][D][sensor:093]: 'r4850 Input voltage': Sending state nan V with 1 decimals of accuracy
[08:51:30][D][sensor:093]: 'r4850 Input current': Sending state nan A with 2 decimals of accuracy
[08:51:30][D][sensor:093]: 'r4850 Input temperature': Sending state nan °C with 1 decimals of accuracy
[08:51:30][D][sensor:093]: 'r4850 Input frequency': Sending state nan Hz with 3 decimals of accuracy
[08:51:30][D][sensor:093]: 'r4850 Output power': Sending state nan W with 1 decimals of accuracy
[08:51:30][D][sensor:093]: 'r4850 Output current': Sending state nan A with 2 decimals of accuracy
[08:51:30][D][sensor:093]: 'r4850 Output voltage': Sending state nan V with 2 decimals of accuracy
[08:51:30][D][sensor:093]: 'r4850 Output temperature': Sending state nan °C with 1 decimals of accuracy
[08:51:30][D][sensor:093]: 'r4850 Efficiency': Sending state nan % with 0 decimals of accuracy
[08:51:30][D][number:012]: 'r4850 Max output current': Sending state nan
[08:51:30][W][component:204]: Component huawei_r4850 took a long time for an operation (0.09 s).
[08:51:30][W][component:205]: Components should block for at most 20-30ms.
[08:51:30][D][sensor:093]: 'r4850 Input power': Sending state 0.00000 W with 1 decimals of accuracy
[08:51:50][D][huawei_r4850:049]: Sending request message
[08:51:50][D][sensor:093]: 'r4850 Input power': Sending state nan W with 1 decimals of accuracy
[08:51:50][D][sensor:093]: 'r4850 Input voltage': Sending state nan V with 1 decimals of accuracy
[08:51:50][D][sensor:093]: 'r4850 Input current': Sending state nan A with 2 decimals of accuracy
[08:51:50][D][sensor:093]: 'r4850 Input temperature': Sending state nan °C with 1 decimals of accuracy
[08:51:50][D][sensor:093]: 'r4850 Input frequency': Sending state nan Hz with 3 decimals of accuracy
[08:51:50][D][sensor:093]: 'r4850 Output power': Sending state nan W with 1 decimals of accuracy
[08:51:50][D][sensor:093]: 'r4850 Output current': Sending state nan A with 2 decimals of accuracy
[08:51:50][D][sensor:093]: 'r4850 Output voltage': Sending state nan V with 2 decimals of accuracy
[08:51:50][D][sensor:093]: 'r4850 Output temperature': Sending state nan °C with 1 decimals of accuracy
[08:51:50][D][sensor:093]: 'r4850 Efficiency': Sending state nan % with 0 decimals of accuracy
[08:51:50][D][number:012]: 'r4850 Max output current': Sending state nan
[08:51:50][W][component:204]: Component huawei_r4850 took a long time for an operation (0.09 s).
[08:51:50][W][component:205]: Components should block for at most 20-30ms.
[08:51:50][D][sensor:093]: 'r4850 Input power': Sending state 0.00000 W with 1 decimals of accuracy
[08:52:10][D][huawei_r4850:049]: Sending request message
[08:52:10][D][sensor:093]: 'r4850 Input power': Sending state nan W with 1 decimals of accuracy
[08:52:10][D][sensor:093]: 'r4850 Input voltage': Sending state nan V with 1 decimals of accuracy
[08:52:10][D][sensor:093]: 'r4850 Input current': Sending state nan A with 2 decimals of accuracy
[08:52:10][D][sensor:093]: 'r4850 Input temperature': Sending state nan °C with 1 decimals of accuracy
[08:52:10][D][sensor:093]: 'r4850 Input frequency': Sending state nan Hz with 3 decimals of accuracy
[08:52:10][D][sensor:093]: 'r4850 Output power': Sending state nan W with 1 decimals of accuracy
[08:52:10][D][sensor:093]: 'r4850 Output current': Sending state nan A with 2 decimals of accuracy
[08:52:10][D][sensor:093]: 'r4850 Output voltage': Sending state nan V with 2 decimals of accuracy
[08:52:10][D][sensor:093]: 'r4850 Output temperature': Sending state nan °C with 1 decimals of accuracy
[08:52:10][D][sensor:093]: 'r4850 Efficiency': Sending state nan % with 0 decimals of accuracy
[08:52:10][D][number:012]: 'r4850 Max output current': Sending state nan
[08:52:10][W][component:204]: Component huawei_r4850 took a long time for an operation (0.09 s).
[08:52:10][W][component:205]: Components should block for at most 20-30ms.

**Comment by mb-software** [comment]
I'm sorry, there is still some documentation missing in this repository.
So far I have used and tested this component only with the ESP32 CAN platform and a sn65hvd230 tranceiver.

In theory it should work also with the mcp2515 platform, but I currently have no hardware to test this. I had to make some small  modifications (increase receive buffer) also on the ESP32 CAN platform because the huawei sends many telegrams at once. I guess there could be a similar issue with the mcp2515.
